tests/e2e/etcd_release_upgrade_test.go

```diff
@@ -165,3 +169,101 @@ func TestReleaseUpgradeWithRestart(t *testing.T) {
 
 	require.NoError(t, ctlV3Get(cx, []string{kvs[0].key}, []kv{kvs[0]}...))
 }
+
+func TestClusterUpgradeAfterPromotingMembers(t *testing.T) {
+	if !fileutil.Exist(e2e.BinPath.EtcdLastRelease) {
+		t.Skipf("%q does not exist", e2e.BinPath.EtcdLastRelease)
+	}
+
+	e2e.BeforeTest(t)
+
+	currentVersion, err := e2e.GetVersionFromBinary(e2e.BinPath.Etcd)
+	require.NoErrorf(t, err, "failed to get version from binary")
+
+	lastClusterVersion, err := e2e.GetVersionFromBinary(e2e.BinPath.EtcdLastRelease)
+	require.NoErrorf(t, err, "failed to get version from last release binary")
+
+	epc := createNewClusterByPromotingMembers(t, e2e.LastVersion)
+	defer func() {
+		require.NoError(t, epc.Close())
+	}()
+
+	err = e2e.DowngradeUpgradeMembers(t, nil, epc, 3, false, lastClusterVersion, currentVersion)
+	require.NoError(t, err)
+}
+
+func createNewClusterByPromotingMembers(t *testing.T, clusterVersion e2e.ClusterVersion) *e2e.EtcdProcessCluster {
```

Comment:
I think we can add one more parameter `clusterSize int`. The first member is always a voting member, and adding all other members as learner one by one.

Reply:
Sounds good. Updated!